auth_matrix/admin.py

```diff
@@ -32,22 +36,23 @@ class Meta:
         )
         export_order = fields
 
-    def dehydrate(self, user):
-        data = super().dehydrate(user)
-        group_names = [group.name for group in Group.objects.all()]
-        for group_name in group_names:
-            data[group_name] = group_name in [group.name for group in user.groups.all()]
+    def after_export(self, queryset, data, *args, **kwargs):
+        group_names_subquery = (
+            Group.objects.filter(user=OuterRef("pk"))
+            .values("user")
+            .annotate(group_names=ArrayAgg("name"), default=Value([]))
+            .values("group_names")
+        )
+        queryset = queryset.annotate(group_names=Subquery(group_names_subquery))
+        for group_name in Group.objects.values_list("name", flat=True).order_by("name"):
+            column = [group_name in (user.group_names or []) for user in queryset]
+            data.append_col(column, header=group_name)
         return data
 
-    def export(self, queryset, *args, **kwargs):
-        dataset = super().export(queryset, *args, **kwargs)
-        group_names = [group.name for group in Group.objects.all()]
-        for group_name in group_names:
-            dataset.append_col(
-                [user.groups.filter(name=group_name).exists() for user in queryset],
-                header=group_name,
-            )
-        return dataset
+
+class UserExportMixin(ExportMixin):
+    resource_classes = (UserGroupResource,)
+    change_list_template = "admin/auth/user/change_list.html"
```

Comment:
IMO the library should not specify what the `change_list_template` is, because this conflicts if other libraries do this.

django-import-export even has an FAQ entry about this: https://django-import-export.readthedocs.io/en/latest/faq.html#failed-to-assign-change-list-template-attribute-warning-in-logs

Proper design is that the library offers the necessary template fragments that the project making use of the library can (and should) then `{% include ... %}` themselves, and that should be part of the installation instructions. So those installation instructions need a code example of how the admin template should be specified and which snippet needs to be included.

Reply:
so you think I should remove it and include this ?

```rst
8. Include the necessary template fragments in your admin templates to enable the authorization matrix functionality. For example, if you override the default User or Group model, you can add this template to your template model's folder (e.g. ``templates/admin/auth/group/change_list.html``):

   .. code-block:: django
      {% extends "admin/change_list.html" %}

      {% load admin_list i18n %}

      {% block object-tools-items %}
          {% include "auth_matrix/admin/includes/object_tools_items.html" %}
          {{ block.super }}
      {% endblock %}

9. Config the template location in the mixin with the matching template.

    .. code-block:: python 
    
       class GroupExportMixin(ExportMixin):
           resource_classes = (GroupPermissionResource,)
           change_list_template = "admin/auth/group/change_list.html"
```

Reply:
How about this ? https://github.com/django-import-export/django-import-export/issues/1514#issuecomment-1344200867

Reply:
Nvm, it makes the export button disappear this way. I can't find a way to re include it. Tried to include the default template location (admin/import_export/change_list_export.html)   within the custom template.